These headers should be legacy to ensure autotools/meson intall the same files.

Upstream-Status: Submitted [https://gitlab.freedesktop.org/xorg/proto/xorgproto/merge_requests/12]
Signed-off-by: Ross Burton <ross.burton@intel.com>

diff --git a/include/X11/extensions/meson.build b/include/X11/extensions/meson.build
index 1d85cf8..d1ac281 100644
--- a/include/X11/extensions/meson.build
+++ b/include/X11/extensions/meson.build
@@ -65,8 +65,6 @@ install_headers(
     'xf86dga.h',
     'xf86dgaproto.h',
     'xf86dgastr.h',
-    'xf86misc.h',
-    'xf86mscstr.h',
     'xf86vm.h',
     'xf86vmproto.h',
     'xf86vmstr.h',
@@ -85,13 +83,6 @@ install_headers(
     'xtestext1const.h',
     'xtestext1proto.h',
     'xtestproto.h',
-    'xtrapbits.h',
-    'xtrapddmi.h',
-    'xtrapdi.h',
-    'xtrapemacros.h',
-    'xtraplib.h',
-    'xtraplibp.h',
-    'xtrapproto.h',
     'Xv.h',
     'XvMC.h',
     'XvMCproto.h',
@@ -113,7 +104,16 @@ if get_option('legacy') == true
         'windowswmstr.h',
         'xcalibrateproto.h',
         'xcalibratewire.h',
+        'xtrapbits.h',
+        'xtrapddmi.h',
+        'xtrapdi.h',
+        'xtrapemacros.h',
+        'xtraplib.h',
+        'xtraplibp.h',
+        'xtrapproto.h',
         'Xeviestr.h',
+        'xf86misc.h',
+        'xf86mscstr.h',
         'xf86rush.h',
         'xf86rushstr.h',
         'XKBgeom.h',
diff --git a/include/X11/meson.build b/include/X11/meson.build
index 1c33c64..a4b022e 100644
--- a/include/X11/meson.build
+++ b/include/X11/meson.build
@@ -59,4 +59,6 @@ install_headers(
 subdir('dri')
 subdir('extensions')
 subdir('fonts')
-subdir('PM')
+if get_option('legacy') == true
+    subdir('PM')
+endif
diff --git a/meson.build b/meson.build
index cfbaa2c..68e622a 100644
--- a/meson.build
+++ b/meson.build
@@ -42,18 +42,15 @@ pcs = [
         ['renderproto',         '0.11.1'],
         ['resourceproto',       '1.2.0'],
         ['scrnsaverproto',      '1.2.2'],
-        ['trapproto',           '3.4.3'],
         ['videoproto',          '2.3.3'],
         ['xcmiscproto',         '1.2.2'],
         ['xextproto',           '7.3.0'],
         ['xf86bigfontproto',    '1.2.0'],
         ['xf86dgaproto',        '2.1'],
         ['xf86driproto',        '2.1.1'],
-        ['xf86miscproto',       '0.9.3'],
         ['xf86vidmodeproto',    '2.3.1'],
         ['xineramaproto',       '1.2.1'],
         ['xproto',              '7.0.32'],
-        ['xproxymngproto',      '1.0.3'],
 ]
 
 foreach pc : pcs
@@ -78,9 +75,12 @@ if get_option('legacy') == true
         ['fontcacheproto', '0.1.3'],
         ['lg3dproto', '5.0'],
         ['printproto', '1.0.5'],
+        ['trapproto', '3.4.3'],
         ['windowswmproto', '1.0.4'],
         ['xcalibrateproto', '0.1.0'],
+        ['xf86miscproto', '0.9.3'],
         ['xf86rushproto', '1.2.2'],
+        ['xproxymngproto', '1.0.3'],
     ]
     foreach pc : legacy_pcs
         pkg.generate(
